modin/pandas/concat.py

```diff
@@ -108,8 +110,15 @@ def concat(
             new_idx_labels = {
                 k: v.index if axis == 0 else v.columns for k, v in zip(keys, objs)
             }
-            tuples = [(k, o) for k, obj in new_idx_labels.items() for o in obj]
+            tuples = [
+                (k, *o) if isinstance(o, tuple) else (k, o)
+                for k, obj in new_idx_labels.items()
+                for o in obj
+            ]
             new_idx = pandas.MultiIndex.from_tuples(tuples)
+            old_name = _determine_name(objs, axis)
+            if old_name is not None:
+                new_idx.names = [None] + old_name
```

Comment:
Why `[None] +`?

Reply:
this is a default name of `keys` level if `names` did not specify, but thanks to your comment I found out that custom `names` does not consider when computing `new_idx`. Fixed that in the last commit